### PR TITLE
WT-13602 Make eviction_progress and dirty target/trigger private.

### PR DESCRIPTION
**Pull request by Siddhartha8899**
These are all the public fields in WT_EVICT.

WT_EVICT -  Setter & Getter | WT_EVICT -  Getter Only
-- | --
eviction_progress | app_waits
evict_max_page_size | app_evicts
evict_max_ms | evict_pass_gen
reentry_hs_eviction_ms | eviction_target
eviction_dirty_target | eviction_trigger
eviction_dirty_trigger | eviction_checkpoint_target
eviction_scrub_target | walk_tree
pass_intr

As part of this PR, the following is done - 
- `eviction_progress`, `eviction_dirty_target` and `eviction_dirty_trigger`has been made private.
- `eviction_trigger`, `eviction_target`, `app_waits`, and `app_evicts`: The use of these fields outside eviction module will be removed, as they are only used externally in cache pool which will be removed soon - [WT-13595](https://jira.mongodb.org/browse/WT-13595) - so no action required.
- For `evict_max_ms`,  `evict_max_page_size `, `reentry_hs_eviction_ms `, `pass_intr`, `walk_tree`, `evict_pass_gen`, `eviction_scrub_target`, and `eviction_checkpoint_target`: These are complex and will be handled as a seperate ticket after a full understanding of each field.

### CODE REVIEW

File: src/btree/bt_discard.c
Comment (ajmorton) [inline src/btree/bt_discard.c:1]: Reading through the PR description the following fields are listed in the table but not addressed in the dot points: 
- `evict_max_ms`
- `evict_max_page_size`
- `reentry_hs_eviction_ms`
How are we handling them?

**Review by ajmorton — APPROVED** [review]
LGTM 👍 

I see the following changes in this PR:
- `__wt_cache_page_evict` has been renamed to `__wt_evict_page_cache_bytes_decr` and moved into `evict_inline.h`. No changes to the function body
- We used to directly set evict thresholds in `__conn_close` and now there's a sub-function to do so with a good warning comment.

One suggestion and one question. Neither need re-review

File: src/btree/bt_discard.c
Comment (Siddhartha8899) [reply src/btree/bt_discard.c:1]: Oops, I forgot to mention these fields in the description but they fall under the third category - complex and will be handled as a seperate ticket after a full understanding of each field.

I wanted something that was an easy win as part of this PR.

File: src/evict/evict_inline.h [hunk 1]
```diff
@@ -260,6 +324,26 @@ __wt_evict_needed(WT_SESSION_IMPL *session, bool busy, bool readonly, double *pc
     return (clean_needed || updates_needed || (!busy && dirty_needed));
 }
 
+/*
+ * __wt_evict_favor_clearing_dirty_cache --
+ *     !!! This function aggressively adjusts the eviction settings to encourage the removal of
+ *     dirty bytes from the cache. Use with caution, as it may significantly impact cache behavior.
```
[the remark this answers is not in the document]
Comment (Siddhartha8899) [reply src/evict/evict_inline.h]: Updated [9e53e92](https://github.com/wiredtiger/wiredtiger/pull/11104/commits/9e53e92a20de442fdef59fdd04be300653f0e6ec)

**Pull request merged**